src/merge.rs

```diff
@@ -40,6 +40,10 @@ pub struct StructureCounts {
     remote_deletes: u64,
 }
 
+/// Holds (matching remote dupes for local GUIDs, matching local dupes for
+/// remote GUIDs).
+type MatchingDupes<'t> = (HashMap<String, Node<'t>>, HashMap<String, Node<'t>>);
```

Comment:
This applies many places, and I'm just leaving the comment here since it's one example, but I almost wonder if it would be worth using `Rc<str>` than all the Strings for the ids, so many of which would be copies of each other...